scylla/src/transport/metrics.rs

```diff
@@ -0,0 +1,92 @@
+use histogram::Histogram;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::{Arc, Mutex};
+
+const ORDER_TYPE: Ordering = Ordering::Relaxed;
+
+pub struct Metrics {
+    errors_num: AtomicU64,
+    queries_num: AtomicU64,
+    errors_iter_num: AtomicU64,
+    queries_iter_num: AtomicU64,
+    histogram: Arc<Mutex<Histogram>>,
+}
+
+impl Metrics {
+    pub fn new() -> Metrics {
+        Self {
+            errors_num: AtomicU64::new(0),
+            queries_num: AtomicU64::new(0),
+            errors_iter_num: AtomicU64::new(0),
+            queries_iter_num: AtomicU64::new(0),
+            histogram: Arc::new(Mutex::new(Histogram::new())),
+        }
+    }
+
+    /// Increments counter for errors that occured in single-page queries.
+    pub fn log_error(&self) {
```

Comment:
nitpick: let's make the name less generic, e.g. `inc_successful_unpaged_queries`. `log_` suggests that we're logging some human-readable messages, which is not exactly the case.

Reply:
Weren't we suppose to count every query that we're trying to execute? `inc_unpaged_queries` would be better if so.